gc/gc-base/src/main/java/org/projectnessie/gc/base/GCParams.java

```diff
@@ -70,6 +70,12 @@ default double getBloomFilterFpp() {
     return 0.03d;
   }
 
+  /** Nessie catalog name to be used with spark to create the output results table. */
+  String getNessieCatalogName();
+
+  /** Output table name to be used for storing the results in default reference. */
+  String getOutputTableIdentifier();
```

Comment:
using the default reference, because if we open a configuration for reference name,

a) what if the reference not exist ?
b) If need to implicitly create reference, what is the the source reference and its hash?
c) Using  SQL for query and insert. So, use reference sql need to be used. Which requires dependency on nessie sql extensions. What if user forgot to configure it ?

Reply:
So every little activity causes unintended conflicts on the "main" branch and "pollute" the commit log of the "main" branch.
I don't really like that approach, because it bugs users with maintenance operations that they haven't actively triggered and very likely cause a lot questions about what that is.

Reply:
hmmm, good point👍🏻.
I thought about conflict before, but as in nessie conflict is per content, I thought conflict will not happen if each time GC creates a new table and other ref may not use gc_table. (but that is a wrong assumption, other ref can have this gc_table name)

I will expose a mandatory config for refName for GC, 
a) If reference does not exist, reference is created pointing to NO_ANCESTOR hash.
b) If reference exist, then use that reference.
c) SQL syntax need ref, so mandatory dependency on NessieSparkSessionExtensions